### PR TITLE
Tools: updates to build_binaries.sh (use waf)

### PR DESCRIPTION
**Pull request by tridge**
this switches to waf for all the px4 builds, plus adds px4pro It also
removes apm1/apm2 for plane build, and removes the download directory
for PX4IO

### CODE REVIEW

**Comment by OXINARF** [comment]
Great to see we are ready for this! If our official binaries start being built by waf, we should definitely add the make deprecation warning to all boards (instead of just Linux) and remove it in 6 months. Once this is in, I'll also remove the slow make builds from CI.

File: Tools/scripts/build_binaries.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # script to build the latest binaries for each vehicle type, ready to upload
 # Andrew Tridgell, March 2013
 
-export PATH=$PATH:/bin:/usr/bin
+export PATH=$HOME/prefix/bin:$PATH:/bin:/usr/bin
```
Comment (lucasdemarchi) [inline Tools/scripts/build_binaries.sh]: Is this really correct? It's usually `$HOME/bin`

File: Tools/scripts/build_binaries.sh [hunk 2]
```diff
@@ -358,21 +314,23 @@ build_arducopter() {
...
+                echo "Failed build of ArduCopter$framesuffix PX4 $tag for $v"
+                error_count=$((error_count+1))
+                continue
+            }
+            cp -f $BUILDROOT/px4-$v/bin/arducopter"$framesuffix".px4 ArduCopter-$v.px4 || {
+                echo "Failed build copy of ArduCopter$framesuffix PX4 $tag for $v"
                 error_count=$((error_count+1))
                 continue
             }
+            copyit ArduCopter-$v.px4 $ddir $tag
```
Comment (lucasdemarchi) [inline Tools/scripts/build_binaries.sh]: why do we keep this in a separate loop from the Linux boards?  aerofc-v1 is built there together with the Linux boards and add a .px4 suffix where needed.

File: Tools/scripts/build_binaries.sh (hunk 1 above)
Comment (tridge) [inline Tools/scripts/build_binaries.sh]: we use $HOME/prefix/bin on the build server

File: Tools/scripts/build_binaries.sh (hunk 2 above)
Comment (tridge) [inline Tools/scripts/build_binaries.sh]: just because it was an easier conversion, and more obviously correct. We could combine them in a later PR

**Comment by tridge** [comment]
@OXINARF I'm not sure we should remove the make build yet. I do want waf in our release builds, as that is what most devs use, but I don't want the make build removed unless we get an ack from all the key devs using windows builds that the waf build works for them. For example, Leonard had trouble with windows waf build last I heard. I don't mind continuing to maintain make build if it helps Leonard.

**Comment by OXINARF** [comment]
@tridge I said 6 months 🙂 I have waf working well on Windows, I'm just not using our Pixhawk toolchain. If we don't put a deprecation warning and we don't tell people that we'll remove it soon, people won't try waf and we'll never fix existing problems - which will mean never removing the Make build system.

File: Tools/scripts/build_binaries.sh (hunk 2 above)
Comment (lucasdemarchi) [inline Tools/scripts/build_binaries.sh]: ok

**Comment by tridge** [comment]
I'll merge this now - we can re-work to simplify the script the way @lucasdemarchi suggested later, and the discussion of deprecating make build can be elsewhere

**Pull request merged**